paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentSheetLoader.kt

```diff
@@ -78,47 +78,119 @@ internal class DefaultPaymentSheetLoader @Inject constructor(
     private val userFacingLogger: UserFacingLogger,
 ) : PaymentSheetLoader {
 
+    @Suppress("LongMethod")
     override suspend fun load(
         initializationMode: PaymentSheet.InitializationMode,
         paymentSheetConfiguration: PaymentSheet.Configuration,
         isReloadingAfterProcessDeath: Boolean,
         initializedViaCompose: Boolean,
     ): Result<PaymentSheetState.Full> = withContext(workContext) {
-        eventReporter.onLoadStarted(initializedViaCompose)
+        runCatching {
+            coroutineScope {
```

Comment:
Needing to nest inside it's own `CoroutineScope` otherwise any failures fail the whole outer scope and are not caught by `runCatching`.

Can make `runCatching` the top-level call with `withContext` as the inner call but that means running `reportFailedLoad` in the scope that `load` was called in.